onnxruntime/core/framework/bfc_arena.cc

```diff
@@ -165,12 +186,13 @@ Status BFCArena::Extend(size_t rounded_bytes) {
   LOGS_DEFAULT(INFO) << "Extended allocation by " << bytes << " bytes.";
 
   stats_.total_allocated_bytes += bytes;
+  stats_.num_arena_extensions += 1;
   LOGS_DEFAULT(INFO) << "Total allocated bytes: "
```

Comment:
Do we need this and next_allocation_region_id_ given it looks like they'll always have the same value?

Reply:
They will - only we are relying on the stats data structure for a counter that could possibly just be owned by BfcArena. (What if we disable stats collection in some build flavor ? ).

I removed next_allocation_region_id_ for now and just used num_arena_extensions